client/src/main/java/org/spine3/client/CommandFactory.java

```diff
@@ -48,48 +70,210 @@
      *
      * @param message the command message
      * @return new command instance
+     * @throws ConstraintViolationThrowable if the passed message does not satisfy the constraints
+     *                                      set for it in its Protobuf definition
      */
-    public Command create(Message message) {
+    public Command create(Message message) throws ConstraintViolationThrowable {
         checkNotNull(message);
+        checkValid(message);
+
         final CommandContext context = createContext();
-        final Command result = Commands.createCommand(message, context);
+        final Command result = createCommand(message, context);
         return result;
     }
 
     /**
-     * Creates command context for a new command.
-     */
-    CommandContext createContext() {
-        return Commands.createContext(actorRequestFactory.getTenantId(),
-                                      actorRequestFactory.getActor(),
-                                      actorRequestFactory.getZoneOffset());
-    }
-
-    /**
      * Creates new {@code Command} with the passed message and target entity version.
      *
      * <p>The command contains a {@code CommandContext} instance with the current time.
      *
+     * <p>The message passed is validated according to the constraints set in its Protobuf
+     * definition. In case the message isn't valid, an {@linkplain ConstraintViolationThrowable
```

Comment:
As discussed vocally, there is an issue #482. I have added the `ValidationException` as another suggestion for renaming.